functional/provision/setup.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+USER_ID=${SUDO_UID:-$(id -u)}
+HOME=$(getent passwd "${USER_ID}" | cut -d: -f6)
+
+export GOROOT=${HOME}/go
+export PATH=${HOME}/go/bin:${PATH}
+
+if [ ! -f ${HOME}/go1.5.3.linux-amd64.tar.gz ]; then
+  # Remove unfinished archive when you press Ctrl+C
```

Comment:
this seems unnecessary, why not just download it in a mktemp and extract it into HOME?